apps/calendar/src/components/events/horizontalEvent.spec.tsx

```diff
@@ -341,4 +366,18 @@ describe('isReadOnly', () => {
 
     expect(showDetailPopupSpy).toHaveBeenCalled();
   });
+
+  it(`should fire 'afterRenderEvent' when it is in readonly mode`, () => {
+    // Given
+    const eventBus = new EventBusImpl<ExternalEventTypes>();
+    const handler = jest.fn();
+    eventBus.on('afterRenderEvent', handler);
+    const { props } = setup();
+
+    // When (resizingWidth)
+    render(<HorizontalEvent {...props} />, { eventBus });
+
+    // Then
+    expect(handler).toBeCalled();
+  });
```

Comment:
This test should belong to the `Firing 'afterRenderEvent'` block.

https://github.com/nhn/tui.calendar/blob/main/apps/calendar/src/components/events/horizontalEvent.spec.tsx#L77

And though other tests should be changed too, how about changing the assertion to the `toHaveBeenCalledTimes(1)`?

Reply:
To put this code in the 'afterRenderEvent' block. You must create a read-only setup function 
like:
```ts
 function setupReadOnly() {
    const eventBus = new EventBusImpl<ExternalEventTypes>();
    const handler = jest.fn();
    eventBus.on('afterRenderEvent', handler);
    const uiModel = new EventUIModel(
      new EventModel({
        id: '1',
        start: new Date(2020, 0, 1, 10, 0),
        end: new Date(2020, 0, 1, 12, 0),
         isReadOnly: true,
      })
    );
    const props = {
      uiModel,
      eventHeight: 30,
      headerHeight: 0,
    };

    return {
      props,
      eventBus,
      handler,
    };
  }
```
It's okay?

Reply:
it's a good idea to change it to `toHaveBeenCalledTimes(1) `
 I'll change it.

Reply:
> To put this code in the 'afterRenderEvent' block. I should create a read-only setup function like:
> 
> ```ts
>  function setupReadOnly() {
>     const eventBus = new EventBusImpl<ExternalEventTypes>();
>     const handler = jest.fn();
>     eventBus.on('afterRenderEvent', handler);
>     const uiModel = new EventUIModel(
>       new EventModel({
>         id: '1',
>         start: new Date(2020, 0, 1, 10, 0),
>         end: new Date(2020, 0, 1, 12, 0),
>          isReadOnly: true,
>       })
>     );
>     const props = {
>       uiModel,
>       eventHeight: 30,
>       headerHeight: 0,
>     };
> 
>     return {
>       props,
>       eventBus,
>       handler,
>     };
>   }
> ```
> 
> It's okay?

Hmm.. Not bad idea. But only the new test uses setup function. I would extract it when I have to write another test for this case.